### PR TITLE
Ref #4879: VPN Churn Improvements - Callout Logic and Settings Changes

### LINKED ISSUE #4879: VPN Billing Retry Changes
Details can be followed in https://github.com/brave/brave-ios/pull/8040

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

This is 3rd and last part of churn improvements for receipt retry state. It is using the verifyReceipt Data method implemented in https://github.com/brave/brave-ios/pull/7803 and also adding logic to show update payment callout for update payment status the callouts are added with https://github.com/brave/brave-ios/pull/7670

This pull request fixes #4879

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

PART 1: Active subscription

- Purchase VPN for 1 month on sandbox account - check it is working properly
- Kill the Browser and ran again 
- In details you should be able to see VPN subscription is active

PART 2: Stop auto-renewable

- Go Appstore -> Subscriptions -> Sandbox Account -> Manage
- Toggle off Allow Purchases & Renewals
- Press done
- This will stop renewing subscription and will put app in grace period

PART 3: Test if Retry logic with grave period is performing (which it doesnt)

- Run the app again
- This time app should be in grace period of some sort after subscription expires (~5min) and you should ne able to see grace period elements in UI
-Custom Alert in the beginning will be shown for devices under iOS 16

## Screenshots:

IMG_3037

https://github.com/brave/brave-ios/assets/6643505/6cda1f6e-2124-413d-91f1-1774452a6f27

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift [hunk 1]
```diff
@@ -255,12 +255,23 @@ extension BrowserViewController {
...
+      
+      if receiptStatus != BraveVPN.ReceiptResponse.Status.retryPeriod.rawValue {
+        return
+      }
     }
     
     presentVPNChurnPromoCallout(for: .updateBillingExpired) {
-      // TODO: Action
+      // Opens Apple's 'manage subscription' screen
+      guard let url = URL(string: "https://apps.apple.com/account/subscriptions") else { return }
```
Comment (iccub) [inline Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift]: Maybe it's time to export this url to a shared constant? It is used in the vpn settings screen as well

Comment (soner-yuksel) [reply Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Callout.swift]: Makes sense, adde new category, fixed at https://github.com/brave/brave-ios/pull/8040/commits/c03a4a93ea069a6bb9a076b7442e09809f318308

File: Sources/Brave/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -1101,12 +1101,8 @@ public class BrowserViewController: UIViewController {
     }
   }
 
-  private func checkCrashRestorationOrSetupTabs() {
-    if crashedLastSession {
```
Comment (iccub) [inline Sources/Brave/Frontend/Browser/BrowserViewController.swift]: removed while debugging?

**Review by iccub — CHANGES_REQUESTED** [review]

File: Sources/Brave/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (soner-yuksel) [reply Sources/Brave/Frontend/Browser/BrowserViewController.swift]: Sorry for mistake, fixed

**Review by iccub — APPROVED** [review]

**Pull request merged**